test/test/api/ApiTests.java

```diff
@@ -11,25 +11,33 @@
 
 public class ApiTests {
 
-    @Test(mainClass = DumpCollapsed.class, jvmArgs = "-Djava.library.path=build/lib", output = true)
+    @Test(mainClass = DumpCollapsed.class, output = true)
     public void flat(TestProcess p) throws Exception {
         Output out = p.waitForExit(TestProcess.STDOUT);
         assert out.contains("BusyLoops.method1;");
         assert out.contains("BusyLoops.method2;");
         assert out.contains("BusyLoops.method3;");
     }
 
-    @Test(mainClass = DumpOtlp.class, jvmArgs = "-Djava.library.path=build/lib")
+    @Test(mainClass = DumpOtlp.class)
     public void otlp(TestProcess p) throws Exception {
         p.waitForExit();
         assert p.exitCode() == 0;
     }
 
-    @Test(mainClass = StopResume.class, jvmArgs = "-Djava.library.path=build/lib", output = true)
+    @Test(mainClass = StopResume.class, output = true)
     public void stopResume(TestProcess p) throws Exception {
         Output out = p.waitForExit(TestProcess.STDOUT);
         assert !out.contains("BusyLoops.method1");
         assert out.contains("BusyLoops.method2");
         assert !out.contains("BusyLoops.method3");
     }
+
+    // https://github.com/async-profiler/async-profiler/issues/1564
+    @Test(mainClass = PrintProfilerVersion.class, output = true)
+    public void version(TestProcess p) throws Exception {
+        String actual = p.waitForExit(TestProcess.STDOUT).toString().trim();
+        String expected = System.getenv("PROFILER_VERSION");
```

Comment:
It's a bad test - at least, in its current shape.
A test must be self-sufficient, it must not depend on variables set outside. First, who guarantees that a the environment variable has the correct value? Second, if a test is run standalone, it will immediately crash with NPE.
We cannot require users/developers/vendors to run the test from Makefile only. Even in our company async-profiler tests are run by custom scripts bypassing Makefile.

Reply:
> Should we even have this test?

That’s exactly what I was getting at :) I think we don't need it - for the reasons you've mentioned.